Source/Scene/ModelInstanceCollection.js

```diff
@@ -0,0 +1,844 @@
+     * per-instance properties like model matrix, pick id, etc.
+     *
+     * @alias ModelInstanceCollection
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {String} [options.url] The url to the .gltf file.
+     * @param {Object} [options.headers] HTTP headers to send with the request.
+     * @param {Object|ArrayBuffer|Uint8Array} [options.gltf] The object for the glTF JSON or an arraybuffer of Binary glTF defined by the CESIUM_binary_glTF extension.
+     * @param {String} [options.basePath=''] The base path that paths in the glTF JSON are relative to.
+     * @param {Boolean} [options.dynamic] Collection is set to stream instance data every frame.
+     * @param {Boolean} [options.show=true] Determines if the collection will be shown.
+     * @param {Boolean} [options.allowPicking=false] When <code>true</code>, each glTF mesh and primitive is pickable with {@link Scene#pick}.
+     * @param {Boolean} [options.asynchronous=true] Determines if model WebGL resource creation will be spread out over several frames or block until completion once all glTF files are loaded.
+     * @param {Boolean} [options.debugShowBoundingVolume=false] For debugging only. Draws the bounding sphere for the collection.
+     * @param {Boolean} [options.debugWireframe=false] For debugging only. Draws the instances in wireframe.
+     *
+     * @exception {DeveloperError} Must specify either <options.gltf> or <options.url>, but not both.
+     * @exception {DeveloperError} Shader program cannot be optimized for instancing. Parameters cannot have any of the following semantics: MODEL, MODELINVERSE, MODELVIEWINVERSE, MODELVIEWPROJECTIONINVERSE, MODELINVERSETRANSPOSE.
+     *
+     * @private
+     */
+    var ModelInstanceCollection = function(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(options.gltf) && !defined(options.url)) {
+            throw new DeveloperError('Either options.gltf or options.url is required.');
+        }
+
+        if (defined(options.gltf) && defined(options.url)) {
+            throw new DeveloperError('Cannot pass in both options.gltf and options.url.');
+        }
+        //>>includeEnd('debug');
+
+        this._instancingSupported = false;
+        this._dynamic = defaultValue(options.dynamic, false);
+        this._show = defaultValue(options.show, true);
+        this._allowPicking = defaultValue(options.allowPicking, true);
+        this._cull = defaultValue(options.cull, true);
+        this._ready = false;
+        this._readyPromise = when.defer();
+        this._state = LoadState.NEEDS_LOAD;
+
+        var instances = defaultValue(options.instances, []);
+        var length = instances.length;
+        var batchData = options.batchData;
+        if (!defined(batchData)) {
+            batchData = new Cesium3DTileBatchData(this, length);
```

Comment:
It depends. I kept it here if `ModelInstanceCollection` is not created from a 3d-tile. But it doesn't really fit...